### PR TITLE
[]mproto.Field -> []*query.Field

### PR DESCRIPTION
**Pull request by sougou**
This part 1 of the QueryResult revamp. With this change,
vitess does not see any MySQL types. The BSON layer still
sends mysql types for backward compatibility, which will
be deleted once deprecated.

### CODE REVIEW

**Comment by alainjobart** [comment]
LGTM. Kinda impossible to review line by line though. :)

File: go/vt/binlog/event_streamer.go [hunk 1]
```diff
@@ -232,10 +232,10 @@ func parsePkTuple(tokenizer *sqlparser.Tokenizer, insertid int64, fields []mprot
 
 			// update type
 			switch fields[index].Type {
-			case mproto.VT_DECIMAL:
+			case sqltypes.Null:
 				// we haven't updated the type yet
-				fields[index].Type = mproto.VT_LONGLONG
-			case mproto.VT_LONGLONG:
+				fields[index].Type = sqltypes.Int64
```
Comment (alainjobart) [inline go/vt/binlog/event_streamer.go]: This one and the next one should be Uint64.

Comment (alainjobart) [inline go/vt/binlog/event_streamer.go]: Never mind this comment, this is wrong in the code, but I don't think this is the time to fix this.

**Comment by sougou** [comment]
@alainjobart 
PTAL. Let me know if you want to submit this, or if I should wait for the gorpc changes. I'm fine either way.

**Comment by alainjobart** [comment]
Go ahead, I have a couple small fixes to do on my side.

**Comment by alainjobart** [comment]
and LGTM obviously.

**Pull request merged**